### DISCUSSION TITLE
Base object name on values of inputs?

### DISCUSSION THREAD

**Discussion by r00key** [pimcore/pimcore#11230] [Q&A / Support]
Hello,

Is a way to base the object names (the one in the tree) on the values of inputs?

For example: I have a "Product" class with two inputs named "ID" and "Name". I make a Product Object with ID = "1234" and Name = "Foo". I would like the object in the tree to automatically be named "1234 Foo" on saving.

(If this could be based on regionalized inputs, that would be really awesome. Also see my related question [11228](https://github.com/pimcore/pimcore/discussions/11228) )

**Comment by passioneight** [comment]
You can create an `EventSubscriber` and listen for the [PRE_UPDATE event](https://github.com/pimcore/pimcore/blob/e953700200a8279cc833bb498592e1a14245df1d/lib/Event/DataObjectEvents.php#L52). From there you can call `setKey` on the element and use the values of your fields.
